nsds_lab_to_nwb/components/stimulus/stim_value_extractor.py

```diff
@@ -2,67 +2,116 @@
 
 
 class StimValueExtractor():
-    def __init__(self, stim_values_command, stim_lib_path):
-        self.stim_values_command = stim_values_command
+    def __init__(self, stim_configs, stim_lib_path):
+        self.stim_name, self.stim_info = check_stimulus_name(stim_configs['name'])
+        self.stim_type = stim_configs['type']
+        self.stim_values_command = stim_configs.get('stim_values', None)
         self.stim_lib_path = stim_lib_path
 
     def extract(self):
-        stim_values_command = self.stim_values_command
-        if 'tone_stimulus_values' in stim_values_command:
-            extractor, filename = self.__parse_command(stim_values_command)
-            if extractor != 'tone_stimulus_values':
-                raise ValueError('parsing error')
-            stim_values = eval('{}(\'{}\')'.format(extractor,
-                            os.path.join(self.stim_lib_path, filename)))
-        if 'timit_stimulus_values' in stim_values_command:
-            extractor, filename = self.__parse_command(stim_values_command)
-            if extractor != 'timit_stimulus_values':
-                raise ValueError('parsing error')
-            stim_values = eval('{}(\'{}\')'.format(extractor,
-                            os.path.join(self.stim_lib_path, filename)))
-        if 'gen_tone_stim_vals' in stim_values_command:
-            extractor, _ = self.__parse_command(stim_values_command)
-            if extractor != 'gen_tone_stim_vals':
-                raise ValueError('parsing error')
-            stim_values = eval('{}()'.format(extractor))
-        if 'np.ones' in stim_values_command:
-            stim_values = eval(stim_values_command)
-
-        return stim_values
-
-    def __parse_command(self, command):
-        ''' return (a_a_a, b.b.b) by parsing string 'a_a_a(b.b.b)' '''
-        res = re.match('(\S+)\((\S+)\)', command)
-        return (res.group(1), res.group(2))
+        if self.stim_type == 'continuous':
+            # this feature is only for discrete stimuli; skip
+            return None
 
+        # for synthetic stimuli
+        if self.stim_values_command is not None:
+            stim_values_command = self.stim_values_command
+            if 'gen_tone_stim_vals' in stim_values_command:
+                # for test_tone_stim
+                stim_values = gen_tone_stim_vals()
+            if 'np.ones' in stim_values_command:
+                # for white noises
+                stim_values = eval(stim_values_command)
+            return stim_values
+
+        # for tone and timit stimuli
+        if 'tone' in self.stim_name:
+            stim_values_path = self._get_stim_values_path()
+            stim_values = tone_stimulus_values(stim_values_path)
+            return stim_values
+        if 'timit' in self.stim_name:
+            stim_values_path = self._get_stim_values_path()
+            stim_values = timit_stimulus_values(stim_values_path)
+            return stim_values
```

Comment:
Can these be unified? I find the first pattern sort of odd (maybe a MARS holdover). Feel free to ignore in this PR and I'll make an issue.

In my opinion, I'd change this logic to
1. `self.stim_name` standardization from the yaml
2. equality checks for `self.stim_name` in if/elif block
3. call the appropriate function.

Reply:
Totally agree that the first pattern is not ideal (remnants of the old mars code, just kept because they work with the current stimulus metadata which is also not ideal)! Your logic sounds good.